waku/node/jsonrpc/relay/handlers.nim

```diff
@@ -37,20 +37,25 @@ type
 
 ## Waku Relay JSON-RPC API
 
-proc installRelayApiHandlers*(node: WakuNode, server: RpcServer, cache: MessageCache) =
+proc installRelayApiHandlers*(node: WakuNode, initPubsubTopics: seq[PubsubTopic], initContentTopics: seq[ContentTopic],server: RpcServer, cache: MessageCache) =
   if node.wakuRelay.isNil():
     debug "waku relay protocol is nil. skipping json rpc api handlers installation"
     return
 
   let topicHandler = proc(topic: PubsubTopic, message: WakuMessage) {.async.} =
       cache.addMessage(topic, message)
 
-  # The node may already be subscribed to some topics when Relay API handlers
-  # are installed
-  for topic in node.wakuRelay.subscribedTopics:
-    node.subscribe(topic, topicHandler)
-    cache.subscribe(topic)
+  let autoHandler = proc(topic: PubsubTopic, message: WakuMessage) {.async.} =
+    if cache.isSubscribed(message.contentTopic):
```

Comment:
Would it be possible to handle the logic of "isSubscribed" from within the `node` instance?

Reply:
I don't see how. `node` has no knowledge of `cache`

`waku_relay` handler is called with a pubsub topic and a message which then wrap this handler here.

How would you filter by content topic?